gpustack/exporter/exporter.py

```diff
@@ -0,0 +1,244 @@
+                            "model_name": model.name,
+                            "runtime": model.backend,
+                            "runtime_version": model.backend_version,
+                            "source": model.source,
+                            "source_key": model.model_source_key,
+                        },
+                    )
+
+                    model_desired_instances.add_metric(
+                        model_label_values,
+                        model.replicas,
+                    )
+
+                    model_running_instances.add_metric(
+                        model_label_values,
+                        model.ready_replicas,
+                    )
+
+                    # instance metrics
+                    instances = model.instances
+                    for mi in instances:
+                        worker_id = str(mi.worker_id) if mi.worker_id else "unknown"
+                        worker_name = mi.worker_name if mi.worker_name else "unknown"
+                        mi_label_values = cluster_label_values + [
+                            worker_id,
+                            worker_name,
+                            str(model.id),
+                            model.name,
+                            mi.name,
+                            mi.state,
+                        ]
+                        model_instance_status.add_metric(
+                            mi_label_values,
+                            1,
+                        )
+
+                # user token used metrics
+                users = await User.all(session)
+                user_id_to_name = {str(user.id): user.username for user in users}
+
+                statement = select(
+                    ModelUsage.user_id,
+                    ModelUsage.model_id,
+                    func.sum(ModelUsage.prompt_token_count).label('prompt_tokens'),
+                    func.sum(ModelUsage.completion_token_count).label(
+                        'completion_tokens'
+                    ),
+                    func.sum(ModelUsage.request_count).label('total_requests'),
+                ).group_by(ModelUsage.user_id, ModelUsage.model_id)
+
+                results = (await session.exec(statement)).all()
+                for row in results:
+                    user_id = str(row.user_id)
+                    user_name = user_id_to_name.get(user_id, "unknown")
+                    cluster_id = model_id_to_cluster_id.get(
+                        str(row.model_id), "unknown"
+                    )
+                    cluster_name = cluster_id_to_name.get(cluster_id, "unknown")
+                    model_id = str(row.model_id)
+                    model_name = model_id_to_name.get(model_id, "unknown")
+                    prompt_tokens = row.prompt_tokens or 0
+                    generation_tokens = row.completion_tokens or 0
+                    user_token_label_values = [
+                        user_id,
+                        user_name,
```

Comment:
Should we avoid per-user metrics? The cardinality of these label can be high. https://www.cncf.io/blog/2025/07/22/prometheus-labels-understanding-and-best-practices/